fusor-ember-cli/app/routes/subscriptions/select-subscriptions.js

```diff
@@ -20,18 +22,38 @@ export default Ember.Route.extend({
 
         var entitlements = this.store.query('entitlement', {uuid: consumerUUID});
         var pools        = this.store.query('pool',        {uuid: consumerUUID});
+        var subscriptions = this.store.query('subscription', {deployment_id: deploymentId, source: 'added'});
 
-        return Ember.RSVP.Promise.all([entitlements, pools]).then(function(results) {
+        return Ember.RSVP.Promise.all([entitlements, pools, subscriptions]).then(function(results) {
           var entitlementsResults = results[0];
           var allPoolsResults     = results[1];
+          var subscriptionResults     = results[2];
           self.modelFor('subscriptions').set('isAuthenticated', true); // in case go to this route from URL
           allPoolsResults.forEach(function(pool){
               pool.set('qtyAttached', 0); //default for loop
+
               entitlementsResults.forEach(function(entitlement) {
                 if (entitlement.get('poolId') === pool.get('id')) {
                   pool.incrementProperty('qtyAttached', entitlement.get('quantity'));
                 }
               });
+
+              //create Fusor::Subscription records if they don't exist
+              var matchingSubscription = subscriptionResults.filterBy('contract_number', pool.get('contractNumber')).get('firstObject');
+              if (Ember.isBlank(matchingSubscription)) {
+                 var sub = self.store.createRecord('subscription', {'contract_number': pool.get('contractNumber'),
+                                     'product_name': pool.get('productName'),
```

Comment:
@cfchase, this is the main change. I'm creating Fusor::Subscription records on the route in `select-subscriptions` with `quantity_to_add: 0` and then update the qty only by calling `saveSubscription`